backend/app/usecase/search/search_test.go

```diff
@@ -0,0 +1,227 @@
+			shortLinks: shortLinksMap,
+			Query: Query{
+				Query: "http google",
+			},
+			filter: Filter{
+				MaxResults: 2,
+				Resources:  []Resource{ShortLink},
+				Orders:     []order.By{order.ByCreatedTimeASC},
+			},
+			relationUsers:      users,
+			relationShortLinks: userShortLinks,
+			expectedHasErr:     true,
+		},
+		{
+			name:       "search without query",
+			shortLinks: shortLinksMap,
+			Query: Query{
+				User: &entity.User{
+					ID:    "alpha",
+					Email: "alpha@example.com",
+				},
+			},
+			filter: Filter{
+				MaxResults: 2,
+				Resources:  []Resource{ShortLink},
+				Orders:     []order.By{order.ByCreatedTimeASC},
+			},
+			relationUsers:      users,
+			relationShortLinks: userShortLinks,
+			expectedHasErr:     true,
+		},
+		{
+			name:       "valid search",
+			shortLinks: shortLinksMap,
+			Query: Query{
+				Query: "http google",
+				User: &entity.User{
+					ID:    "alpha",
+					Email: "alpha@example.com",
+				},
+			},
+			filter: Filter{
+				MaxResults: 2,
+				Resources:  []Resource{ShortLink},
+				Orders:     []order.By{order.ByCreatedTimeASC},
+			},
+			relationUsers:      users,
+			relationShortLinks: userShortLinks,
+			expectedResult: Result{
+				shortLinks: []entity.ShortLink{
+					{
+						Alias:    "google",
+						LongLink: "https://google.com",
+					},
+					{
+						Alias:    "git-google",
+						LongLink: "http://github.com/google",
+					},
+				},
+				users: nil,
+			},
+		},
+		{
+			name:       "empty search",
+			shortLinks: shortLinksMap,
+			Query: Query{
+				Query: "non existent",
+				User: &entity.User{
+					ID:    "alpha",
+					Email: "alpha@example.com",
+				},
+			},
+			filter: Filter{
+				MaxResults: 2,
+				Resources:  []Resource{ShortLink},
+				Orders:     []order.By{order.ByCreatedTimeASC},
+			},
+			relationUsers:      users,
+			relationShortLinks: userShortLinks,
+			expectedResult: Result{
+				shortLinks: nil,
+				users:      nil,
+			},
+		},
+		{
+			name:       "one result",
+			shortLinks: shortLinksMap,
+			Query: Query{
+				Query: "short",
+				User: &entity.User{
+					ID:    "alpha",
+					Email: "alpha@example.com",
+				},
+			},
+			filter: Filter{
+				MaxResults: 2,
+				Resources:  []Resource{ShortLink},
+				Orders:     []order.By{order.ByCreatedTimeASC},
+			},
+			relationUsers:      users,
+			relationShortLinks: userShortLinks,
+			expectedResult: Result{
+				shortLinks: []entity.ShortLink{
+					{
+						Alias:    "short",
+						LongLink: "https://short-d.com",
+					},
+				},
+				users: nil,
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+			userShortLinkRepo := repository.NewUserShortLinkRepoFake(testCase.relationUsers, testCase.relationShortLinks)
+			shortLinkRepo := repository.NewShortLinkFake(testCase.shortLinks)
+			timeout := time.Second
+
+			search := NewSearch(&shortLinkRepo, &userShortLinkRepo, timeout)
```

Comment:
This is a meta comment, but if timeouts are being used `context.Context` might be a possibility in the future.

Reply:
Definitely! As of now, the search request is canceled only if some resource is taking a long time to return the result. In my opinion, for now, we should leave its due to its low complexity and if the complexity increases then it would be rather easy to change.